SIL.BuildTasks.Tests/FileUpdateTests.cs

```diff
@@ -62,6 +63,120 @@ public void GetModifiedContents_RegexTextNotMatched_Throws(string origContents,
+			Assert.That(result, Is.EqualTo(expectedResult));
+		}
+
+		[TestCase("_DATE_ _VERSION_\r\nStuff", "M/yyyy", "{0} 3.2.1\r\nStuff")]
+		[TestCase("_DATE_ _VERSION_\r\nStuff done before _DATE_", "dd-MM-yy", "{0} 3.2.1\r\nStuff done before {0}")]
+		public void GetModifiedContents_SpecialDatePlaceholderButFileDoesNotSpecifyFormat_InsertsDateWithSpecifiedDateFormat(
+			string origContents, string dateFormat, string expectedResultFormat)
+		{
+			var updater = new FileUpdate
+			{
+				Regex = "_VERSION_",
+				ReplacementText = "3.2.1",
+				DatePlaceholder = "_DATE(*)_",
+				DateFormat = dateFormat
+			};
+			
+			var currentDate = DateTime.UtcNow.Date.ToString(dateFormat ?? updater.DateFormat);
+
+			var result = updater.GetModifiedContents(origContents);
+			var expectedResult = string.Format(expectedResultFormat, currentDate);
+			Assert.That(result, Is.EqualTo(expectedResult));
+		}
+
+		[TestCase("MM-yy")]
+		[TestCase("dd MMMM")]
+		public void GetModifiedContents_SpecialDatePlaceholderWithFileSpecifyingFormat_InsertsDateWithFormatFromFile(
+			string format)
+		{
+			var origContents = $"_DATE({format})_\r\nStuff";
+					
+			var updater = new FileUpdate
+			{
+				Regex = "(.*)",
+				ReplacementText = "$1",
+				DatePlaceholder = "_DATE(*)_",
+			};
+
+			var currentDate = DateTime.UtcNow.Date.ToString(format);
+
+			var result = updater.GetModifiedContents(origContents);
+			Assert.That(result, Is.EqualTo($"{currentDate}\r\nStuff"));
+		}
+
+		[TestCase("MM-yyyy", "d MMMM yy")]
+		[TestCase("dd MMMM", "MM/dd/yyyy")]
+		public void GetModifiedContents_SpecialDatePlaceholderWithFileSpecifyingMultipleFormats_InsertsDateWithFormatsFromFile(
+			string format1, string format2)
+		{
+			var origContents = $"First _DATE({format1})_\r\nSecond _DATE_\r\nLast _DATE({format2})_";
+
+			var updater = new FileUpdate
+			{
+				Regex = "(.*)",
+				ReplacementText = "$1",
+				DatePlaceholder = "_DATE(*)_",
+			};
+
+			var currentDate1 = DateTime.UtcNow.Date.ToString(format1);
+			var currentDateInDefaultFmt = DateTime.UtcNow.Date.ToString(updater.DateFormat);
+			var currentDate2 = DateTime.UtcNow.Date.ToString(format2);
+
+			var result = updater.GetModifiedContents(origContents);
+			Assert.That(result, Is.EqualTo($"First {currentDate1}\r\nSecond {currentDateInDefaultFmt}\r\nLast {currentDate2}"));
+		}
+
+		[TestCase("es")]
+		[TestCase("fr")]
+		public void GetModifiedContents_SpecialDatePlaceholderWithLocalizedFileSpecifyingFormat_InsertsLocaleSpecificDateWithFormatFromFile(string locale)
+		{
+			var origContents = "_DATE(d MMMM yyyy)_\r\nStuff";
+
+			var updater = new FileUpdate
+			{
+				File = $"ReleaseNotes.{locale}.md",
+				FileLocalePattern = @"\.(?<locale>[a-z]{2}(-\w+)?)\.md$",
+				Regex = "(.*)",
+				ReplacementText = "$1",
+				DatePlaceholder = "_DATE(*)_",
+			};
+
+			var currentDate = string.Format(DateTime.UtcNow.Date.ToString("d {0} yyyy"),
+				GetMonthName(locale, DateTime.UtcNow.Month));
```

Comment:
This would be slightly easier:

```suggestion
			var currentDate = DateTime.UtcNow.Date.ToString("d MMMM yyyy", new CultureInfo(locale));
```

Reply:
True. I was intentionally trying to make the test be as much as possible NOT a replication of the production code and trying to illustrate clearly in the test what the point of the locale logic is. I was considering when I wrote GetMonthName the possibility of implementing it by hard-coding the month names.

Reply:
With the use of the ReproducibleDateTimeProvider, I was able to just hardcode the expected month names.
<!-- Reviewable comment -ORHWmQSDGcTToJAc_m2:b9ypcrd -->
<!-- Sent from Reviewable.io -->